src/singer_clojure/messages.clj

```diff
@@ -0,0 +1,83 @@
+(ns singer-clojure.messages
+  (:require [clojure.data.json :as json]
+            [clojure.test :refer [is]]))
+
+(defn valid?
+  "Function to evalulate whether a singer message is valid"
+  [message]
+  (and (#{"SCHEMA" "STATE" "RECORD" "ACTIVATE_VERSION"} (message "type"))
+       (case (message "type")
+         "SCHEMA"
+         (message "schema")
+
+         "STATE"
+         (message "value")
+
+         "RECORD"
+         (message "record")
+
+         "ACTIVATE_VERSION"
+         (message "version"))))
+
+(defn write!
+  "Writes a valid singer message to stdout"
+  [message]
+  {:pre [(is (valid? message) (format "message %s is not valid" message))]}
+  (-> message
+      (json/write-str)
+      println))
+
+(defn make-unsupported-schemas-empty [schema-message catalog stream-name]
```

Comment:
Implementation details like this could probably be marked private (`defn-`)

Reply:
Fixed